Mathlib/Order/SupClosed.lean

```diff
@@ -0,0 +1,269 @@
+/-
+Copyright (c) 2023 Yaël Dillies, Christopher Hoskin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Christopher Hoskin
+-/
+import Mathlib.Data.Finset.Lattice
+import Mathlib.Order.Closure
+
+/-!
+# Sets closed under join/meet
+
+This file defines predicates for sets closed under `⊔` and shows that each set in a join-semilattice
+generates a join-closed set and that a semilattice where every directed set has a least upper bound
+is automatically complete. All dually for `⊓`.
+
+## Main declarations
+
+* `SupClosed`: Predicate for a set to be closed under join (`a ∈ s` and `b ∈ s` imply `a ⊔ b ∈ s`).
+* `InfClosed`: Predicate for a set to be closed under meet (`a ∈ s` and `b ∈ s` imply `a ⊓ b ∈ s`).
+* `supClosure`: Sup-closure. Smallest sup-closed set containing a given set.
+* `infClosure`: Inf-closure. Smallest inf-closed set containing a given set.
+* `SemilatticeSup.toCompleteSemilatticeSup`: A join-semilattice where every sup-closed set has a
+  least upper bound is automatically complete.
+* `SemilatticeInf.toCompleteSemilatticeInf`: A meet-semilattice where every inf-closed set has a
+  greatest lower bound is automatically complete.
+-/
+
+namespace Finset
+variable {α β : Type*}
+
+-- This will come from https://github.com/leanprover-community/mathlib/pull/18989
```

Comment:
What does this comment mean?

Reply:
It means that this lemma will come from the forward port of this mathlib3 PR. I'm inclined to just leave it there for now since the forward port is imminent and adds more lemmas like that.

Reply:
bors merge